modules/tinymce/src/core/main/ts/fmt/ApplyFormat.ts

```diff
@@ -335,16 +335,16 @@ const applyFormat = (ed: Editor, name: string, vars?: FormatVars, node?: Node |
       if (!isCollapsed || !FormatUtils.isInlineFormat(format) || TableCellSelection.getCellsFromEditor(ed).length) {
         // Apply formatting to selection
         selection.setRng(RangeNormalizer.normalize(selection.getRng()));
-        SelectionUtils.preserve(selection, true, () => {
-          SelectionUtils.runOnRanges(ed, (selectionRng, fake) => {
-            const expandedRng = fake ? selectionRng : ExpandRange.expandRng(ed, selectionRng, formatList);
-            applyRngStyle(dom, expandedRng, false);
+
+        FormatUtils.adjustSelectionAfter(ed, () => {
```

Comment:
This is rather messy and hard to read... is they any reason we couldn't move the `SelectionUtils.preserve` logic inside the `FormatUtils` helper and rename it to something like `preserveSelection` as that's what it's really trying to do here? We could then just treat this as a custom wrapper to `SelectionUtils.preserve` for formatting specifically and it could then easily be moved later should we need to.

Reply:
True. That would probably be a cleaner abstraction. Happy to make that change

Reply:
Have made the update. Thanks for the suggestion